packages/ember-intl/addon/services/intl.ts

```diff
@@ -0,0 +1,397 @@
+import { getOwner } from '@ember/application';
+import { registerDestructor } from '@ember/destroyable';
+import { cancel, next, type Timer as EmberRunTimer } from '@ember/runloop';
+import Service from '@ember/service';
+import { htmlSafe } from '@ember/template';
+import { tracked } from '@glimmer/tracking';
+import EventEmitter from 'eventemitter3';
+
+import type {
+  FormatDateParameters,
+  FormatListParameters,
+  FormatMessageParameters,
+  FormatNumberParameters,
+  FormatRelativeParameters,
+  FormatTimeParameters,
+  IntlShape,
+  OnErrorFn,
+} from '../-private/formatjs/index';
+import {
+  createIntl,
+  createIntlCache,
+  formatDate,
+  formatList,
+  formatMessage,
+  formatNumber,
+  formatRelative,
+  formatTime,
+} from '../-private/formatjs/index';
+import { escapeFormatMessageOptions } from '../-private/utils/escape-format-message-options';
+import { getDOM } from '../-private/utils/get-dom';
+import {
+  convertToArray,
+  convertToString,
+  hasLocaleChanged,
+} from '../-private/utils/locale';
+import { flattenKeys, type Translations } from '../-private/utils/translations';
+import translations from '../translations';
+
+type OnFormatjsError = (error: Parameters<OnErrorFn>[0]) => void;
+
+export default class IntlService extends Service {
+  @tracked private _intls: Record<string, IntlShape> = {};
+  @tracked private _locale?: string[];
+
+  private _cache = createIntlCache();
+  private _eventEmitter = new EventEmitter();
+  private _formats?: Record<string, unknown>;
+  private _timer?: EmberRunTimer;
+
+  get locales(): string[] {
+    return Object.keys(this._intls);
+  }
+
+  get primaryLocale(): string | undefined {
+    if (!this._locale) {
+      return;
+    }
+
+    return this._locale[0];
+  }
+
+  constructor() {
+    // eslint-disable-next-line prefer-rest-params
+    super(...arguments);
+
+    if (!this._formats) {
+      // @ts-expect-error: Property 'resolveRegistration' does not exist on type 'Owner'
+      this._formats = getOwner(this).resolveRegistration('formats:main') ?? {};
+    }
+
+    // Hydrate
+    translations.forEach(([locale, translations]: [string, Translations]) => {
+      this.addTranslations(locale, translations);
+    });
```

Comment:
**technical risk**

I'm not sure how this exactly works and what we can do instead to prepare for Embroider.